pkg/operator/k8sutil/prometheus.go

```diff
@@ -86,6 +87,14 @@ func CreateOrUpdatePrometheusRule(prometheusRule *monitoringv1.PrometheusRule) (
 		if !errors.IsAlreadyExists(err) {
 			return nil, fmt.Errorf("failed to create prometheusRules. %+v", err)
 		}
+		// Get current PrometheusRule so the ResourceVersion can be set as needed
+		// for the object update operation
+		promRule, err := client.MonitoringV1().PrometheusRules(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		prometheusRule.ObjectMeta.ResourceVersion = promRule.ObjectMeta.ResourceVersion
```

Comment:
```suggestion
		prometheusRule.ObjectMeta = promRule.ObjectMeta
```

Reply:
It is better that we copy the entire metadata of the existing object (like OwnerReferences) otherwise we'll lose it after Update.

Reply:
The operator regenerates the object meta every time including the owner references so we should only need the `ResourceVersion` field

Reply:
@umangachapagain is that ok for you?

Reply:
Yes.